### DISCUSSION TITLE
why is exclude in task configuration not applied?

### DISCUSSION THREAD

**Discussion by WebTiger89** [detekt/detekt#7589] [Q&A] [answered]
I would like to exclude the build folder but this:
```
tasks.withType<Detekt>().configureEach {
    include(
        "**/*.kt",
        "**/*.kts",
    )
    exclude("**/build/**")
}
```

does nothing. When I comment out exclude I would expect to get warnings for the .kt files in build folder. Is build folder excluded by default?

Also it would be great to log the processed folders when **debug** fals is enabled

**Comment by WebTiger89** [comment]
Well, detekt adds "build/" to exclusion list by default, same for includes. See https://github.com/detekt/detekt/blob/84a59b7c233b384252dcf7cd56d1a5a0b37e6051/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/DetektPlugin.kt#L148

The default can be overriden like:
```
tasks.withType<Detekt>().configureEach {
    // Defaults to '**/*.kt', '**/*.kts'
    include(
        "**/*.kt",
        "**/*.kts",
    )
    setExcludes(listOf())
    println("excludes: $excludes")
    // Defaults to 'build/'
    //exclude("build/")
}
```

**Comment by BraisGabin** [comment, marked as answer]
The exclude function is a gradle functionality. And they only let you exclude based on the relative path. A relative path that doesn't includes the build folder. For that reason it doesn't work.
